vms/avm/vm.go

```diff
@@ -421,22 +452,48 @@ func (vm *VM) Linearize(_ context.Context, stopVertexID ids.ID, toEngine chan<-
 		vm.txBackend,
 		vm.chainManager,
 		&vm.clock,
-		vm.mempool,
+		mempool,
 	)
 
-	vm.network = network.New(
+	vm.network, err = network.New(
 		vm.ctx,
 		vm.parser,
-		vm.chainManager,
-		vm.mempool,
+		network.NewLockedTxVerifier(
+			&vm.ctx.Lock,
+			vm.chainManager,
+		),
+		mempool,
 		vm.appSender,
+		vm.registerer,
+		txGossipHandlerID,
```

Comment:
nit: seems like there are enough args for a config struct

Reply:
I'm considering re-working this to take in options rather than how it is currently structured